samples/nrf_desktop/src/modules/hid_state.c

```diff
@@ -391,52 +391,49 @@ static void connect_subscriber(const void *subscriber_id, bool is_usb)
 
 static void disconnect_subscriber(const void *subscriber_id)
 {
+	struct subscriber *s = NULL;
+
 	for (size_t i = 0; i < ARRAY_SIZE(state.subscriber); i++) {
 		if (subscriber_id == state.subscriber[i].id) {
-			bool is_usb = state.subscriber[i].is_usb;
+			s = &state.subscriber[i];
+			break;
+		}
+	}
 
-			if (&state.subscriber[i] == state.selected) {
-				enum target_report tr = TARGET_REPORT_MOUSE;
-				struct report_state *rs =
-					&state.selected->state[tr];
+	if (s == NULL) {
+		LOG_WRN("Cannot disconnect subscriber");
+		return;
+	}
 
-				if (rs->state != STATE_DISCONNECTED) {
-					/* Clear state if notification was not
-					 * disabled before disconnection.
-					 */
-					struct report_data *rd =
-						&state.report_data[tr];
+	if (s == state.selected) {
+		enum target_report tr = TARGET_REPORT_MOUSE;
+		struct report_state *rs = &s->state[tr];
 
-					LOG_INF("Clear mouse report data");
-					memset(&rd->items, 0, sizeof(rd->items));
-					eventq_reset(&rd->eventq);
-				}
-			}
+		if (rs->state != STATE_DISCONNECTED) {
+			/* Clear state if notification was not
+			 * disabled before disconnection.
+			 */
+			struct report_data *rd = &state.report_data[tr];
 
-			memset(&state.subscriber[i],
-			       0,
-			       sizeof(state.subscriber[i]));
+			LOG_INF("Clear mouse report data");
+			memset(&rd->items, 0, sizeof(rd->items));
+			eventq_reset(&rd->eventq);
+		}
+	}
 
-			LOG_INF("Subscriber %p disconnected", subscriber_id);
+	memset(s, 0, sizeof(*s));
 
-			if (&state.subscriber[i] == state.selected) {
-				state.selected = NULL;
-				if (is_usb) {
-					state.selected =
```

Comment:
This is the logic change I mentioned above. Instead of selecting BLE subscriber only if we were in the USB we are doing it always.
BLE cannot be a selected subscriber if USB is connected. At the moment we also support only one USB connection. So behavior of the program will be the same.
The difference is when we will support multiple BLE connections in which case (if there is no USB) another BLE connection will be selected.